### DISCUSSION TITLE
Would like to contribute

### DISCUSSION THREAD

**Discussion by rmorgan1973** [magiblot/turbo#71] [Ideas]
I would like to contribute so that I can use the editor in my workflow as an IDE.

I'd like to add VS Code like features - project browser, terminal window, etc.  I would like to be able to remotely work on code through a terminal from say an iPad but also support the ability to spawn an app like running "nom run start" on the host system.

I used Turbo Pascal and Turbo C extensively starting from late 80's.
Would you be open to those contributions?  I have not contributed to any Github code so it's a bit new to me for the proper protocol.

**Comment by rmorgan1973** [comment]
Given that I possibly want to take the app in a different direction, perhaps a fork would be better?  Would be nice to at least share some of the features such as setting persistence, open recent files, etc.

**Comment by magiblot** [comment]
Hi!

I would have liked to implement some of these features myself, but I didn't because I lacked the time. Of course I won't deter others from trying to do it, so you have my support. Me too would like this app to have many more features. In fact, one of the reasons I created this editor is to prove the point that it isn't so difficult to create a usable text editor for the terminal. Although the Turbo Vision framework is (in my opinion) neither simple nor easy, I believe that using it still saves you time compared to using ncurses or the like directly, like most people do.

When it comes to persistent configuration, my idea was to use an INI or TOML-based format.

And when it comes to the internal terminal, I suggested that it might be better to instead have a usable independent terminal multiplexer. See #24.

Cheers.

**Reply by rmorgan1973** [reply]
I was thinking about ini or json for the settings so we're on the same page with that.

I agree that tmux is better for a fully interactive terminal.  I was thinking more about an output window that captured stdout and stderr.  Easy way to run a programs/scripts and get results.  

I did have this idea of interactive terminals for some widgets to make almost a TUI desktop.. clock, calendar, weather, top, etc.

Another crazy idea I had is setting your terminal font really, really small and doing low res CGA/EGA/VGA 320x200 graphics with text characters. I wanted to play DOS games on my iPad through ssh + dosemu. I'm weird like that.
